org.civicrm.reqangular/src/common/services/before-hash-query-params.service.js

```diff
@@ -0,0 +1,79 @@
+/* eslint-env amd */
+
+define([
+  'common/modules/services'
+], function (module) {
+  'use strict';
+
+  module.factory('beforeHashQueryParams', ['$log', '$window',
+    function ($log, $window) {
+      $log.debug('Service: beforeHashQueryParams');
+
+      return {
+        parse: parse
+      };
+
+      /**
+       * Parses url into list of query strings
+       *
+       * @param {String} url
+       */
+      function parse (url) {
+        var urlToParse = url || $window.location.href;
+        var cleanUrl = getUrlBeforeHash(urlToParse);
+
+        if (urlToParse.indexOf('?') === -1 || cleanUrl.indexOf('=') === -1) {
+          return {};
+        }
+
+        return prepareQueryParamsObject(getQueryParams(cleanUrl));
+      }
+
+      /**
+       * Prepare Query params object
+       *
+       * @param {Array} queryTokens
+       * @return {Object}
+       */
+      function prepareQueryParamsObject (queryTokens) {
+        var queryParams = {};
+
+        queryTokens.forEach(function (token, index) {
+          var splittedToken = token.split('=');
+
+          if (splittedToken[0]) {
```

Comment:
what if the url was `/civicrm/contact/view/?reset&cid` ?

Reply:
Added the coverage for this case. now in this case the query params object will be:
```js
{ reset: null, cid: null }
```